docs/user_guide/datasets.rst

```diff
@@ -0,0 +1,187 @@
+
+    >>> metrics = {
+    ... 'accuracy': skm.accuracy_score,
+    ... 'precision': skm.precision_score,
+    ... 'recall': skm.recall_score,
+    ... 'false positive rate': false_positive_rate,
+    ... 'true positive rate': true_positive_rate,
+    ... 'selection rate': selection_rate, 
+    ... 'count': count}
+    >>> grouped_metric = MetricFrame(metrics=metrics,
+    ... y_true=y_test, 
+    ... y_pred=predicted,
+    ... sensitive_features=X_test["B"])
+    >>> print(grouped_metric.by_group)
+        accuracy precision    recall false positive rate true positive rate  \
+    B                                                                            
+    False  0.852459      0.92  0.766667            0.064516           0.766667   
+    True   0.863636  0.909091  0.833333                 0.1           0.833333   
+
+        selection rate count  
+    B                           
+    False       0.409836    61  
+    True             0.5    66  
+
+    >>> grouped_metric = MetricFrame(metrics=metrics,
+    ... y_true=y_test,
+    ... y_pred=predicted,
+    ... sensitive_features=X_test["LSTAT"])
+    >>> print(grouped_metric.by_group)
+        accuracy precision    recall false positive rate true positive rate  \
+    LSTAT                                                                        
+    False  0.901639  0.910714  0.980769            0.555556           0.980769   
+    True   0.818182       1.0  0.142857                 0.0           0.142857   
+
+        selection rate count  
+    LSTAT                       
+    False       0.918033    61  
+    True        0.030303    66  
+
+
+.. _discussion:
+
+Discussion
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Boston housing dataset is rife with ethical issues, and in general, we strongly discourage using it. We've kept it in fairlearn because of its potential as a teaching tool for how to deal with ethical issues in a dataset. There are ways to `debias the data <https://scikit-lego.netlify.app/fairness.html>`, but other benchmark datasets exist that do not present these issues.
+
+It's important to keep the differences between the way Harrison and Rubenfield used the dataset and the way modern machine learning practicioners have used it in focus. Harrison and Rubenfield conducted an empirical econometric study, the goal of which was to determine the causal impacts of these variables on median home value. Interpretation of causal models involves looking at model coefficients to ascertain the effect of one variable on the depedent variable, holding all other factors fixed. This use case is different than the typical supervised learning exercise, where the goal is to predict house values using the signals in the dataset. Harrison and Rubenfield also made many choices about what to include in the dataset and how to represent those features, and they do not sufficiently defend their inclusion and transformation of variables in their published work. 
```

Comment:
Again, please highlight the different *consequences* of using a ML model in practice (i.e. as a predictive model to set prices) compared to a descriptive model to understand the past.

I am not 100% convinced by the argument that these variables should not have been included in the study by Harrison and Rubenfield. I am definitely not an expert on housing prices in the 1980's but the assumption that the proportion of Blacks in a neighborhood affect the house prices (unfortunately) does not seem very far-fetched. I think the primary objection to using this dataset for benchmarking is that it implies that the dataset is suitable for predictive modeling, not that the original use of the dataset was necessarily problematic.

Reply:
I think you're correct that what we want to highlight here is the danger of using the dataset for predictive modelling, though I might disagree that the way they used the data in their original use case was sound (they made some methodological choices that are not easily defensible). At any rate, this should be a fast fix and I can make the edits today. Question though, should I make the edits inline here or offline and then re-commit the edited file?

Reply:
I'm definitely not an expert on environmental economics and haven't read the paper - so I'll take your word for it! If you have the time (and interest), could you perhaps give a short summary of which choices were not easily defendable and why? If there's a lesson there that could be relevant to our audience it would be great to include it in one way or another (from the current text it just wasn't immediately obvious to me what that lesson was).

Re. edits: with inline edits, do you mean committing suggested changes or directly editing the file on Github? If it's the former, I'd definitely just leverage Github's functionality to commit right away (if you agree with the suggestion, of course!). If you meant editing on Github vs editing your local copy offline: just go with whatever works best for you. All of these options should result in commits to your fork so for us it doesn't really matter! :)

Reply:
Oh yes, I meant should I edit the file directly on GitHub, but I will commit suggested changes and edit within Github! I just made a commit that addresses this comment and cuts down on the overflow lines so hopefully it's easier to review.